### ISSUE TITLE
Services configured with `PrivateNetwork=yes` fail to start in containers

### ISSUE THREAD

**Issue by enr0n** [systemd/systemd#29526]
### systemd version the issue has been seen with

git main

### Used distribution

Ubuntu 23.10

### Linux kernel version used

6.5.0-5-generic

### CPU architectures issue was seen on

aarch64

### Component

systemd

### Expected behaviour you didn't see

Services configured with `PrivateNetwork=yes` should be able to start inside containers.

### Unexpected behaviour you saw

Services configured with `PrivateNetwork=yes` fail to start within containers. Specifically, services configured this way fail when pid1 is attempting to mount a private sysfs for mount namespacing:

```
root@six:~# systemd-detect-virt 
systemd-nspawn
root@six:~# systemd-run -p PrivateNetwork=yes --wait /bin/true
Running as unit: run-u9.service
Finished with result: exit-code
Main processes terminated with: code=exited/status=226
Service runtime: 14ms
CPU time consumed: 14ms
root@six:~# systemctl status run-u9.service
× run-u9.service - /bin/true
     Loaded: loaded (/run/systemd/transient/run-u9.service; transient)
  Transient: yes
     Active: failed (Result: exit-code) since Tue 2023-10-10 17:33:23 EDT; 7s ago
   Duration: 14ms
    Process: 144 ExecStart=/bin/true (code=exited, status=226/NAMESPACE)
   Main PID: 144 (code=exited, status=226/NAMESPACE)
        CPU: 14ms

Oct 10 17:33:23 six (true)[144]: run-u9.service: Failed to set up mount namespacing: /sys: Operation not permitted
Oct 10 17:33:23 six (true)[144]: run-u9.service: Failed at step NAMESPACE spawning /bin/true: Operation not permitted
Oct 10 17:33:23 six systemd[1]: run-u9.service: Child 144 belongs to run-u9.service.
Oct 10 17:33:23 six systemd[1]: run-u9.service: Main process exited, code=exited, status=226/NAMESPACE
Oct 10 17:33:23 six systemd[1]: run-u9.service: Failed with result 'exit-code'.
Oct 10 17:33:23 six systemd[1]: run-u9.service: Service will not restart (restart setting)
Oct 10 17:33:23 six systemd[1]: run-u9.service: Changed running -> failed
Oct 10 17:33:23 six systemd[1]: run-u9.service: Unit entered failed state.
Oct 10 17:33:23 six systemd[1]: run-u9.service: Consumed 14ms CPU time.
Oct 10 17:33:23 six systemd[1]: run-u9.service: Releasing resources...
```

### Steps to reproduce the problem

1. Create a container for use with systemd-nspawn. I did this on Ubuntu 23.10 using debootstrap to create an Ubuntu 23.10 container, but I don't think it really matters how the container is created.

```bash
sudo debootstrap mantic /var/lib/machines/mantic
sudo machinectl start mantic
sudo machinectl shell mantic
```

2. In the container, follow [the instructions](https://systemd.io/HACKING/) to build systemd from source, and then install it inside the container with `ninja -C build install`.
3. Reboot the container.
4. Enter the container again with `machinectl shell mantic`, and attempt to start any service with `PrivateNetwork=yes`, such as `systemd-localed.service` or `systemd-hostnamed.service`. Or, run a transient service:

```bash
systemd-run --wait -p PrivateNetwork=yes /bin/true
``` 

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
root@six:~# journalctl -u run-u9.service --no-pager -b
Oct 10 17:33:23 six systemd[1]: run-u9.service: Failed to load configuration: No such file or directory
Oct 10 17:33:23 six systemd[1]: run-u9.service: Trying to enqueue job run-u9.service/start/fail
Oct 10 17:33:23 six systemd[1]: run-u9.service: Installed new job run-u9.service/start as 624
Oct 10 17:33:23 six systemd[1]: run-u9.service: Enqueued job run-u9.service/start as 624
Oct 10 17:33:23 six systemd[1]: run-u9.service: Will spawn child (service_enter_start): /bin/true
Oct 10 17:33:23 six systemd[1]: run-u9.service: Failed to set 'trusted.invocation_id' xattr on control group /system.slice/run-u9.service, ignoring: Operation not permitted
Oct 10 17:33:23 six systemd[1]: run-u9.service: Failed to remove 'trusted.delegate' xattr flag on control group /system.slice/run-u9.service, ignoring: Operation not permitted
Oct 10 17:33:23 six systemd[1]: run-u9.service: Failed to remove 'trusted.survive_final_kill_signal' xattr flag on control group /system.slice/run-u9.service, ignoring: Operation not permitted
Oct 10 17:33:23 six systemd[1]: run-u9.service: Passing 0 fds to service
Oct 10 17:33:23 six systemd[1]: run-u9.service: About to execute: /bin/true
Oct 10 17:33:23 six systemd[1]: run-u9.service: Forked /bin/true as 144
Oct 10 17:33:23 six (true)[144]: PR_SET_MM_ARG_START failed: Operation not permitted
Oct 10 17:33:23 six (true)[144]: Found cgroup2 on /sys/fs/cgroup/, full unified hierarchy
Oct 10 17:33:23 six (true)[144]: run-u9.service: Kernel keyring access prohibited, ignoring.
Oct 10 17:33:23 six systemd[1]: run-u9.service: Changed dead -> running
Oct 10 17:33:23 six systemd[1]: run-u9.service: Job 624 run-u9.service/start finished, result=done
Oct 10 17:33:23 six systemd[1]: Started run-u9.service.
Oct 10 17:33:23 six (true)[144]: Successfully added address 127.0.0.1 to loopback interface
Oct 10 17:33:23 six (true)[144]: Successfully added address ::1 to loopback interface
Oct 10 17:33:23 six (true)[144]: Successfully brought loopback interface up
Oct 10 17:33:23 six (true)[144]: Bind-mounting / on /run/systemd/mount-rootfs (MS_BIND|MS_REC "")...
Oct 10 17:33:23 six (true)[144]: Applying namespace mount on /run/systemd/mount-rootfs/run/credentials
Oct 10 17:33:23 six (true)[144]: Bind-mounting /run/systemd/inaccessible/dir on /run/systemd/mount-rootfs/run/credentials (MS_BIND|MS_REC "")...
Oct 10 17:33:23 six (true)[144]: Successfully mounted /run/systemd/inaccessible/dir to /run/systemd/mount-rootfs/run/credentials
Oct 10 17:33:23 six (true)[144]: Applying namespace mount on /run/systemd/mount-rootfs/run/systemd/incoming
Oct 10 17:33:23 six (true)[144]: Followed source symlinks /run/systemd/propagate/run-u9.service → /run/systemd/propagate/run-u9.service.
Oct 10 17:33:23 six (true)[144]: Bind-mounting /run/systemd/propagate/run-u9.service on /run/systemd/mount-rootfs/run/systemd/incoming (MS_BIND "")...
Oct 10 17:33:23 six (true)[144]: Successfully mounted /run/systemd/propagate/run-u9.service to /run/systemd/mount-rootfs/run/systemd/incoming
Oct 10 17:33:23 six (true)[144]: Applying namespace mount on /run/systemd/mount-rootfs/sys
Oct 10 17:33:23 six (true)[144]: Failed to umount /run/systemd/mount-rootfs/sys, ignoring: Invalid argument
Oct 10 17:33:23 six (true)[144]: Failed to umount /run/systemd/mount-rootfs/sys, ignoring: Invalid argument
Oct 10 17:33:23 six (true)[144]: Failed to umount /run/systemd/mount-rootfs/sys, ignoring: Invalid argument
Oct 10 17:33:23 six (true)[144]: Failed to umount /run/systemd/mount-rootfs/sys, ignoring: Invalid argument
Oct 10 17:33:23 six (true)[144]: Failed to umount /run/systemd/mount-rootfs/sys, ignoring: Invalid argument
Oct 10 17:33:23 six (true)[144]: Failed to umount /run/systemd/mount-rootfs/sys, ignoring: Invalid argument
Oct 10 17:33:23 six (true)[144]: Failed to umount /run/systemd/mount-rootfs/sys, ignoring: Invalid argument
Oct 10 17:33:23 six (true)[144]: Failed to umount /run/systemd/mount-rootfs/sys, ignoring: Invalid argument
Oct 10 17:33:23 six (true)[144]: Mounting sysfs (sysfs) on /run/systemd/mount-rootfs/sys (MS_NOSUID|MS_NODEV|MS_NOEXEC "")...
Oct 10 17:33:23 six (true)[144]: Failed to mount sysfs (type sysfs) on /run/systemd/mount-rootfs/sys (MS_NOSUID|MS_NODEV|MS_NOEXEC ""): Operation not permitted
Oct 10 17:33:23 six (true)[144]: run-u9.service: Failed to set up mount namespacing: /sys: Operation not permitted
Oct 10 17:33:23 six (true)[144]: run-u9.service: Failed at step NAMESPACE spawning /bin/true: Operation not permitted
Oct 10 17:33:23 six systemd[1]: run-u9.service: Child 144 belongs to run-u9.service.
Oct 10 17:33:23 six systemd[1]: run-u9.service: Main process exited, code=exited, status=226/NAMESPACE
Oct 10 17:33:23 six systemd[1]: run-u9.service: Failed with result 'exit-code'.
Oct 10 17:33:23 six systemd[1]: run-u9.service: Service will not restart (restart setting)
Oct 10 17:33:23 six systemd[1]: run-u9.service: Changed running -> failed
Oct 10 17:33:23 six systemd[1]: run-u9.service: Unit entered failed state.
Oct 10 17:33:23 six systemd[1]: run-u9.service: Consumed 14ms CPU time.
Oct 10 17:33:23 six systemd[1]: run-u9.service: Releasing resources...
```

**Comment by yuwata** [comment]
@enr0n Thank you! I will take a look.

**Comment by yuwata** [comment]
Hmm, I cannot reproduce the issue. The container has special settings for /sys/ ?

**Comment by yuwata** [comment]
Hopefully, fixed by #29529. Could you test the PR?
